### PR TITLE
Ronen branch2

### PR DESCRIPTION
**Pull request by Ronenii**

### CODE REVIEW

File: B23 Ex03 Ronen 319047718 Ido 315942193/GrageUserInterface.cs [hunk 1]
```diff
@@ -9,12 +9,24 @@ public class GrageUserInterface
...
+                printWelcomeMessage();
+                printMenu();
+                printErrorMessage(errorMessage); // TODO: find a way to get exceptions into the errorMessage
+                userInput = getUserChoice();
+                action = Action.ConvertIntgerToeAction(userInput);
+                handleRequestByAction(action);
+                if(action != Action.eAction.Exit)
+                {
+                    Console.ReadKey();
+                }
```
Comment (idobi111) [inline B23 Ex03 Ronen 319047718 Ido 315942193/GrageUserInterface.cs]: insert to handleRequestByAction() and write log there of BYE BYE

File: B23 Ex03 Ronen 319047718 Ido 315942193/GrageUserInterface.cs [hunk 2]
```diff
@@ -71,20 +83,29 @@ private void printMenu()
...
                 isValid = false;
             }
             return isValid;
         }
 
+        private void printErrorMessage(string i_ErrorMessage)
+        {
+            if(i_ErrorMessage != null)
+            {
+                Console.WriteLine(i_ErrorMessage);
```
Comment (idobi111) [inline B23 Ex03 Ronen 319047718 Ido 315942193/GrageUserInterface.cs:112]: Do we want to add Error: ErrorMessage as we did in ex2

File: B23 Ex03 Ronen 319047718 Ido 315942193/GrageUserInterface.cs [hunk 3]
```diff
@@ -71,20 +83,29 @@ private void printMenu()
...
         }
 
-        private bool validateUserActionInput(string i_UserInput, out int o_userInputInteger)
+        private bool validateUserActionInput(string i_UserInput, out int o_UserInputInteger)
         {
             bool isValid = true;
-            int.TryParse(i_UserInput, out o_userInputInteger);
-            if (!(o_userInputInteger > 0 && o_userInputInteger <= 7))
+            int.TryParse(i_UserInput, out o_UserInputInteger);
+            if (!(o_UserInputInteger > 0 && o_UserInputInteger <= 7))
```
Comment (idobi111) [inline B23 Ex03 Ronen 319047718 Ido 315942193/GrageUserInterface.cs:100]: choose >= or only >

File: B23 Ex03 Ronen 319047718 Ido 315942193/GrageUserInterface.cs (hunk 2 above)
Comment (Ronenii) [reply B23 Ex03 Ronen 319047718 Ido 315942193/GrageUserInterface.cs:112]: Think so, looks better if the user gets feedback each time what is wrong with his action

File: B23 Ex03 Ronen 319047718 Ido 315942193/GrageUserInterface.cs (hunk 3 above)
Comment (Ronenii) [reply B23 Ex03 Ronen 319047718 Ido 315942193/GrageUserInterface.cs:100]: I only changed your var naming, you wrote this code

**Comment by Ronenii** [comment]
Commented on questions and resolved what you asked me to

**Review by idobi111 — APPROVED** [review]

**Review by idobi111 — APPROVED** [review]

**Review by idobi111 — APPROVED** [review]

**Pull request merged**